### ISSUE TITLE
[ Current Show Widget ] Does not auto refresh (Pro version)

### ISSUE THREAD

**Issue by montereyjeff** [netmix/radio-station#485]
**Describe the bug**
On my site the widgets for Current Show On-Air and Upcoming Shows do not reload on changeover.

**To Reproduce**
1. Go to https://kspb.org and watch the right sidebar.

**Expected behavior**
Sidebar should update on a show changeover.

**Screenshots**
![Screenshot 2024-02-28 142623](https://github.com/netmix/radio-station/assets/53131566/05b52f47-8a70-43d4-a697-25feee5085c1)

**Additional context**
We're an early user of Radio Station--our Pro version is via the Add-on.

**Comment by tonyzeoli** [comment]
Hey Jeff, you were one of our first customers and we appricate that.

Thanks for reporting this. I do see in the widget that it says, "Upgrade to PRO." Did you check your account panel in the plugin and make sure that the license key is set to activate PRO?

I see you're on 2.6.4, from what your account dashboard shows.

There is a new beta version, 2.6.4.4 that you can take. Please login to your Freemius Dashboard to check the versions and see if that is available to you. If not, we'll have to fix it to make it available to you. This may solve the issue. But I'm adding @majick777 here to get involved to resolve for you quickly.

**Comment by montereyjeff** [comment]
Hey Tony, under Account Details it says Plugin: Radio Station; Version: 2.5.7; Plan: FREEMIUM VERSION. Under Add-Ons it says Radio Station Pro, Version 2.6.4, Plan PROFESSIONAL. The buttons on the right says Deactivate License, Cancel Subscription, Change License, and Sync.

**Comment by montereyjeff** [comment]
I updated the Radio Station Pro plugin to 2.6.4.4. No change to the widgets.

**Comment by kairandles** [comment]
I also have this issue on my site :)

**Comment by tonyzeoli** [comment]
@montereyjeff @kairandles 

@majick777 is looking into this. We'll report back shortly. Your patience is appreciated. Thanks!

**Comment by majick777** [comment]
@montereyjeff @kairandles I did fix an issue with this in the free version 2.5.7 that was preventing the timestamp from being passed correctly for the reloader, can you confirm that you have 2.5.7 installed?

**Comment by montereyjeff** [comment]
@majick777 
> Hey Tony, under Account Details it says Plugin: Radio Station; Version: 2.5.7; Plan: FREEMIUM VERSION. Under Add-Ons it says Radio Station Pro, Version 2.6.4, Plan PROFESSIONAL. The buttons on the right says Deactivate License, Cancel Subscription, Change License, and Sync.

Yes, Free version is 2.5.7; Radio Station Pro Add-on is now 2.6.4.4.

**Comment by tonyzeoli** [comment]
@montereyjeff did you clear any caching after taking the update? Just checking.

**Comment by montereyjeff** [comment]
I did clear the cache, yes.

**Comment by majick777** [comment]
@montereyjeff Sorry for the delay, been down with the flu and catching up on things. 

It looks to me like Siteground Optimizer plugin is combining your scripts, including the Radio Station scripts. But it isn't doing a good job at all! Because there is a whole bunch of inline scripts that are supposed to be printed *after* the main scripts are loaded, as they depend on those functions being available. Instead it is putting the combined main scripts last instead. So the inline scripts are all out of position, and this is going to cause lots of errors, if not with Radio Station then with something else (currently seems to be crashing moment.js and preventing the countdown triggering.)

This should be fairly basic stuff for a script optimizer to get right. You mightbe able to optimize the Siteground Optimizer settings to fix this by adding exclusions, but you might just end up with problems with other scripts anyway. Maybe just turn off their script combining completely, and probably do the same for their CSS combining... Then use another plugin like the free Autoptimize that handles those better (or Perfmatters if you want to pay for the best.) 

You might need to keep Siteground Optimizer activated for their caching, or if they have a separate caching plugin (I can't remember offhand) then just keep the cache one activated and disable Siteground Optimizer in favor of one of the other plugins mentioned.

**Comment by majick777** [comment]
@kairandles Could you send more information on your issue, the above is specific to Jeff's site as he's on Siteground. It's unlikely yours is related, unless by coincidence you are also using Siteground Optimizer.

**Comment by montereyjeff** [comment]
@majick777 I deactivated the SG Optimizer so it's out of the picture now. I've had my station home page open since before the top of the hour and now at :04 after the hour the right-hand widget has not refreshed.

**Comment by kairandles** [comment]
Hey @majick777 this is on youthschoice.co.uk the one on the top menu doesn't but the one on the bottom does!

**Comment by montereyjeff** [comment]
@majick777 It's still not automatically refreshing even after I deactivated SG Optimizer.

**Comment by majick777** [comment]
@montereyjeff Okay, I can see the console errors have gone due to that, but it's still not reloading. 

Sorry for the delay on this it's had me a bit stumped. I had an epiphany though, I believe it's due to the fact that you don't have countdown display enabled on the widget. So I've looked at the code and realized I wrote the reloader as an extension to the countdown... so without the countdown display enabled also, the reloader doesn't do anything, because then the countdown script isn't loaded either. Erk.

To test this, you can enable the countdown on the widget and then add some CSS to hide the display.
`.rs-countdown {display: none;}`

You can also try the latest development version (of Free in this case) where I've added a fix for this.
https://github.com/netmix/radio-station/tree/develop

@kairandles I believe the cause might be similar for you, with one widget having countdown enabled and the other not? I'm not sure as the URL you gave is redirecting to another site and I can't see the widgets there.

**Comment by montereyjeff** [comment]
@majick777 I enabled the countdown. At the top of the hour this happened:
![image](https://github.com/netmix/radio-station/assets/53131566/26e1416a-2ad9-4ec0-a7e8-ee07b1f33cee)

**Comment by tonyzeoli** [comment]
@majick777 I think we lost track of this with @montereyjeff last report way back in March. Can we check this.

**Comment by tonyzeoli** [comment]
@majick777 Let's take one final look at this. Seems resolved but not sure.
